### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by zepheiryan** [chef/opscode-pushy-client#97]
We upgraded our network of nodes to the 2.1.0-1 release of push jobs client following an upgrade to the latest Chef client from 12.4.x (yes, it had been a while).  A couple weeks later, some of our nodes started to fall over due to memory exhaustion.  We have them set up to send the output of top to us when resources start to get short, and all of them would indicate the push jobs client virtual memory was at the 20GB+ mark.  We also have some graphs that show the same, hitting close to 60GB before the graphing fails.

Our limited usage of the client (with push jobs server 1.1.6-1, so we have allow unencrypted enabled) is still working, so it doesn't appear to be driven by errors.   The logging picture for the client appears to have changed and stopped logging to the client log, so I've only just started to get info-level logs for one node to see if it does in fact complain before things go south.  The server logs are usually error free, but sometimes there are complaints about the webmachine failing while servicing requests for pushy config.

I've only just started digging, but I'm fairly certain we're operating with normal parameters and didn't see anything else out there about push jobs and memory leaks, so I figured I'd open up a thread on it.  Let me know if there's more I can do to provide further context or details.

**Comment by zepheiryan** [comment]
I managed to catch one in progress.  I don't have much to show for it other than that `pmap` showed thousands of lines like this:

```
    7fae298da000 rw-p 00000000  00:00      0    2052     24     24         20        24    0      0 [stack:8042]
    7fae29adb000 ---p 00000000  00:00      0       4      0      0          0         0    0      0 
```

whereas `pmap` on a non-misbehaving process returns just a few hundred lines.  Using top, it looks like virtual memory is increasing by around 2k/s, or a pair of those lines every second.  I'm not sure if this matters either, but `lsof` listed two connections to the push jobs server port 10002, whereas other not misbehaving nodes only have one.

**Comment by zepheiryan** [comment]
Appears to be fixed in 2.1.1, so seemingly related to the 2.x vs 1.x issue that that release resolves.
